packages/client/src/create-deployment.ts

```diff
@@ -99,18 +99,33 @@ export default function buildCreateDeployment() {
             entries: fileList.map(file => relative(workPath, file)),
           })
           .pipe(createGzip());
-        const tarBuffer = await streamToBuffer(tarStream);
-        debug('Packed tarball');
-        files = new Map([
-          [
-            hash(tarBuffer),
-            {
-              names: [join(workPath, '.vercel/source.tgz')],
-              data: tarBuffer,
-              mode: 0o666,
-            },
-          ],
-        ]);
+        if (process.env.SPLIT_SOURCE_ARCHIVE) {
```

Comment:
Not a huge fan of doing the env var check directly in `@vercel/client`. Perhaps it should be an option that gets passed in here? Maybe we'll want to allow a CLI flag to trigger the behavior or change the interface in some other way in the future.

Reply:
Also, we usually prefix env vars with `VERCEL_`